### PR TITLE
 [issue #470] 

### PR DESCRIPTION
**Pull request by dipak-pawar**
This Fixes 
1.naming convention of docker-compose services & networks.
2.assigning network mentioned in services to container if mentioned otherwise default.

### CODE REVIEW

File: docker/docker/pom.xml [hunk 1]
```diff
@@ -98,6 +98,11 @@
         </dependency>
 
         <dependency>
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
```
Comment (bartoszmajsak) [inline docker/docker/pom.xml]: `<scope>test</scope>`?

File: docker/docker/src/main/java/org/arquillian/cube/docker/impl/client/DockerContainerDefinitionParser.java [hunk 2]
```diff
@@ -69,20 +69,21 @@ private static DockerCompositions convert(URI uri, DefinitionFormat definitionFo
...
+            String fileContent = "";
             if(uri.isAbsolute()) {
                 content = IOUtil.asStringPreservingNewLines(uri.toURL().openStream());
             } else {
-                String fileContent = uri.toString();
+                fileContent = uri.toString();
                 content = IOUtil.asStringPreservingNewLines(new FileInputStream(fileContent));
             }
-            return convert(content, definitionFormat);
+            return convert(content, definitionFormat, fileContent);
```
Comment (bartoszmajsak) [inline docker/docker/src/main/java/org/arquillian/cube/docker/impl/client/DockerContainerDefinitionParser.java]: Why do we need both `content` and `fileContent`? Isn't it file location?

**Comment by lordofthejars** [comment]
Change the PR title to show what is fixing. Now it is a simple number. You can use the constructor [issue #470] and the description.

File: docker/docker/src/test/java/org/arquillian/cube/docker/impl/docker/compose/DockerComposeConverterTest.java [hunk 3]
```diff
@@ -21,212 +22,246 @@
 
 public class DockerComposeConverterTest {
```
Comment (dipak-pawar) [inline docker/docker/src/test/java/org/arquillian/cube/docker/impl/docker/compose/DockerComposeConverterTest.java]: yes I have run all tests belonging to this test class.

**Comment by dipak-pawar** [comment]
@alexsoto, Tests was creating network but failing due change in network configuration. I have updated tests to new logic. Don't forget to do , mvn install as ftest-docker-compose-v2 is depends on snapshots in project.

**Comment by lordofthejars** [comment]
@dipak-pawar why have you modified these tests? They are running correctly.

**Comment by dipak-pawar** [comment]
@lordofthejars , Following are the reasons:
1.We couldn't access container name using arquillian-resource as actual container is tomcat & our service is running by name rootDir_tomcat.
2.changed port as container is running on custom network, So getting connection refused for 8081.

**Comment by lordofthejars** [comment]
But still I don't understand the reason, in my computer using the master
branch before your change, everything works and a network called frontend
is created, the tests are executed without any problem.

Look:

docker network ls

NETWORK ID          NAME                DRIVER              SCOPE
1080531f9ec6        bridge              bridge              local
de8138d3ec7e        front-tier          bridge              local
80ff8d2201d8        host                host                local
9cbbc4ab1229        none                null                local

docker network inspect de81

[
    {
        "Name": "front-tier",
        "Id":
"de8138d3ec7e42d3f860796040273017e6a9b98269dc76c496a47285f91d1c18",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.18.0.0/16",
                    "Gateway": "172.18.0.1/16"
                }
            ]
        },
        "Internal": false,
        "Containers": {},
        "Options": {},
        "Labels": {}
    }
]

And after the test which passes correctly

docker network ls

NETWORK ID          NAME                DRIVER              SCOPE
1080531f9ec6        bridge              bridge              local
80ff8d2201d8        host                host                local
9cbbc4ab1229        none                null                local

So for this reason I don't understand this PR at all bcause it seems to work


El dj., 13 oct. 2016 a les 16:05, Dipak Pawar (<notifications@github.com>)
va escriure:

**Comment by dipak-pawar** [comment]
@lordofthejars In master branch it's creating network but service is not using created network which is bug.
If you look at output of docker network inspect de81 it should include service container. currently its empty

**Comment by lordofthejars** [comment]
ah true, but then your PR fixes this right? But I cannot see the relation
ship with the port modification you did. Why in frontend network cannot use
8081?

El dj., 13 oct. 2016 a les 16:29, Dipak Pawar (<notifications@github.com>)
va escriure:

**Comment by dipak-pawar** [comment]
@lordofthejars need to figure out. I'll figure it out. No worries.

**Comment by lordofthejars** [comment]
Ok, great. And Thanks for taking care of this, now I understand exactly what is happening, at first I thought the problem was that network was not created, but the problem is that container is not attached

File: docker/docker/src/main/java/org/arquillian/cube/docker/impl/client/DockerContainerDefinitionParser.java [hunk 4]
```diff
@@ -69,20 +69,21 @@ private static DockerCompositions convert(URI uri, DefinitionFormat definitionFo
...
             if(uri.isAbsolute()) {
                 content = IOUtil.asStringPreservingNewLines(uri.toURL().openStream());
             } else {
-                String fileContent = uri.toString();
-                content = IOUtil.asStringPreservingNewLines(new FileInputStream(fileContent));
+                fileLocation = uri.toString();
+                content = IOUtil.asStringPreservingNewLines(new FileInputStream(fileLocation));
             }
-            return convert(content, definitionFormat);
+            return convert(content, definitionFormat, fileLocation);
```
Comment (dipak-pawar) [inline docker/docker/src/main/java/org/arquillian/cube/docker/impl/client/DockerContainerDefinitionParser.java]: @lordofthejars I didn't get what do you want to say?

**Comment by bartoszmajsak** [comment]
Closed by #480

**Pull request closed without merging**